core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -112,13 +112,18 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
-    public static SuggestBuilder fromXContent(QueryParseContext parseContext, Suggesters suggesters) throws IOException {
+    public static SuggestBuilder fromXContent(QueryParseContext parseContext, Suggesters suggesters, final boolean fromStart)
+        throws IOException {
+
         XContentParser parser = parseContext.parser();
         ParseFieldMatcher parseFieldMatcher = parseContext.parseFieldMatcher();
         SuggestBuilder suggestBuilder = new SuggestBuilder();
         String fieldName = null;
 
         XContentParser.Token token;
+        if (fromStart && parser.nextToken() != XContentParser.Token.START_OBJECT) {
```

Comment:
@cbuescher @areek I added this `fromStart` method parameter to differentiate, when parsing from x-content, between the situation where the suggest is part of a larger search request, and when the suggest is on its own as part of the suggest API.  This may go away later when we merge the implementation of the search with suggest and standalone suggest into one, as @areek suggested before, but what I noticed was in some places, the parsing was failing and always returning back "suggestion name not found", even though there may have been other issues with the parsing.  Also, valid JSON was not being parsed correctly by the suggest API.  The reason is because in the search with suggest scenario, it has already parsed the starting token for the suggest object, whereas in the suggest API scenario, the starting token has not yet been parsed.  In some places, we were adding another `parser.nextToken()` to account for this, but it wasn't being applied consistently and it was error prone to manage that.  I feel having the method itself have a parameter that declares the intentions of the parsing (whether we have already parsed the starting `{` or not) is more clear and less error prone, but I welcome your thoughts.